### ISSUE TITLE
package.json engines.node when using object rest/spread syntax

### ISSUE THREAD

**Issue by coreyfarrell** [sindresorhus/meta#8]
I wasn't sure where to ask so this seemed like a reasonable place.  What do you feel is an appropriate value for engines.node in a package which uses object rest/spread syntax?

```js
this.options = {
  ...defaultOptions,
  ...options
};
```

This feature was introduced in node.js 8.3.0 by the upgrade of v8 to 6.0.  LTS started at 8.9.0.  So when `engines.node` is set to `>=8` this is technically incorrect.  Sorry I didn't notice this before posting patches to `make-dir` and a couple others.

Refs:
[node.js 8.3.0 changelog](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V8.md#8.3.0)
[v8 6.0 release](https://v8.dev/blog/v8-release-60#object-rest%2Fspread-properties)

**Comment by sindresorhus** [comment]
IMHO, it's expected that people upgrade patch versions of Node.js. Node.js point versions outside LTS are also technically considered unsupported when the major version goes into LTS.

**Comment by coreyfarrell** [comment]
Agreed on the expectation that pre-LTS point versions are unsupported.  One possible issue is if `xo` ever enables the rule `node/no-unsupported-features/es-syntax`.  That rule would throw an error that object rest/spread is unsupported for node.js 8.0.0.  Not sure you ever plan to enable that rule, I know it's problematic for ava tests and anything else which uses `@babel/plugin-env`.

**Comment by rafamel** [comment]
Hi there, thanks for the swift response to [#8.](https://github.com/sindresorhus/read-pkg-up/issues/8)

I just wanted to note that there are ramifications to this. Specifically, when running containerized environments for testing on node@8, and idem for CIs. Most people do set node versions to the first release, meaning, in order to use this package with those tools, they will need to specifically set their tools to match the required node version, which, most importantly, is **not** node@8, but a minor version of it.

For me, personally, that would be reason enough to use `Object.assign` instead, but there's also the spec issue. This is just my take on this, but I would not assume what people will or won't do according to what, in my view, they *should* do or should have done. If a package declares node@8 as their node version, I will count on it to work in any minor or patch -someone might have or not have reasons for not upgrading, but that's besides the point. The issue is, there's a contract there, which is independent of my assumptions on whether someone should have node@^8.3.0 or not. Again, this is just my take on this, but I would put forwards that, if it doesn't actually work in node@8.0.0, it shouldn't be declared as its node major version -or it should actually work on it.

So, to recap, I'm not sure how many there are out there taking this hard approach, but for me, if I want to support node@8 in my own packages -which I do- that would mean that I can only have your libraries as dependencies when their node version is lower than 8 -so I can count on them to actually work on node@8.

Hope it makes sense.

**Comment by coreyfarrell** [comment]
Just to be clear I was never suggesting that any code should be changed.  I raised this as a potential documentation issue.  I disagree with any effort to take the code backwards.

**Comment by sindresorhus** [comment]
> ne possible issue is if xo ever enables the rule node/no-unsupported-features/es-syntax.

This rule is now enabled, so I'll be notified when I need to upgrade the minor version in package.json.
